### PR TITLE
Add initial unit tests and CI pipeline

### PR DESCRIPTION
**Pull request by dvdgomez**
Hey folks, this PR establishes the initial unit tests and CI pipeline for slurmrestd-operator.

### Summary of Change

- Add initial templates for bug reports, enhancement requests, and pull requests
- Add a basic CI pipeline using GitHub actions
              - Inclusive language check
              - Lint
                  - A separate PR will cover linting changes, currently linting has not been run yet. Those changes will stem from the results of running `tox -e lint`.
              - Unit tests
- Formatting
              - Changes to the source code in this PR stem from running `tox -e fmt`. These changes result from using `black` and `ruff`.
- Unit Tests
              - These tests only test the typical expected behavior of the charm. Integration and functional tests will follow in later PRs however those are not covered here. 

Please take a look and ensure these changes are satisfactory and raise any questions, comments, or concerns!

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -79,7 +76,7 @@ def _on_fluentbit_relation_created(self, event):
 
     def _configure_fluentbit(self):
         logger.debug("## Configuring fluentbit")
-        cfg = list()
```
Comment (matheushent) [inline src/charm.py:82]: I'm the one who advocates for the `list()` or `dict()` or whatever definition. I'm curious about the reason to change it :)

**Review by jaimesouza — APPROVED** [review]
Looks good to me! Good work!

File: src/charm.py (hunk 1 above)
Comment (matheushent) [reply src/charm.py:82]: my concern is more related to readability. We have had people in the team with some sort of dyslexia, so, `[]` or `{}` was difficult to read sometimes, that's why we always have preffered `list()` over `[]` for example

Comment (matheushent) [reply src/charm.py:82]: but I do think you have a good argument. Let's move on

**Comment by jamesbeedy** [comment]
Going to merge this now, as all comments are resolved. Thanks @dvdgomez!

**Pull request merged**